### PR TITLE
Add <Table /> molecule.

### PR DESCRIPTION
**Pull request by ConnorBryan**
https://mycryptobuilds.com/ui/add-table/?selectedKind=Molecules&selectedStory=Table

### CODE REVIEW

File: src/molecules/Table/Table.tsx [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+  offset?: number;
+}
+
+export interface TableConfig {
+  sortableColumn: string | null;
+  sortFunction?(a: any, b: any): number;
+}
+
+export interface TableContent {
+  body: any[][];
```
[the remark this answers is not in the document]
Comment (ConnorBryan) [reply src/molecules/Table/Table.tsx]: Addressed.

File: src/molecules/Table/Table.tsx [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+  border-top: 0.0625em solid ${props => props.theme.tableHeadBorder};
+  border-bottom: 0.0625em solid ${props => props.theme.tableHeadBorder};
+  background: ${props => props.theme.tableHeadBackground};
+` as any;
+
+const TableHeading = styled(Typography)`
+  ${sharedCellProperties}
+  color: ${props => props.theme.headline};
+  text-align: left;
+  font-size: 0.875em;
```
[the remark this answers is not in the document]
Comment (ConnorBryan) [reply src/molecules/Table/Table.tsx]: Addressed.

File: src/molecules/Table/Table.tsx [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+    const { collapsedGroups, sortedColumnDirection } = this.state;
+    const { body, groups = [] } = this.getSortedLayout();
+
+    return (
+      <table {...rest}>
+        <thead>
+          <TableHead>
+            {head.map((heading, index) => {
+              const isSortableColumn =
+                (config as TableConfig).sortableColumn === heading;
```
[the remark this answers is not in the document]
Comment (ConnorBryan) [reply src/molecules/Table/Table.tsx]: Addressed.

File: src/molecules/Table/Table.tsx [hunk 4]
```diff
@@ -0,0 +1,322 @@
...
+  // Potentially reverse the new order depending on the sort direction.
+  const finalSortedColumnIndices =
+    sortedColumnDirection === ColumnDirections.Forward
+      ? sortedColumnIndices
+      : sortedColumnIndices.reverse();
+  // Apply the new order to all of the rows.
+  const sortedRows = finalSortedColumnIndices.map(index => body[index]);
+
+  return sortedRows;
+};
```
[the remark this answers is not in the document]
Comment (ConnorBryan) [reply src/molecules/Table/Table.tsx]: Addressed.

File: src/molecules/Table/index.ts [hunk 5]
```diff
@@ -1 +1 @@
-export { default } from './Table';
+export { default as Table } from './Table';
```
Comment (nickmccurdy) [inline src/molecules/Table/index.ts]: Sorry I meant this line should go in `src/modules/index.ts`, and this file should be left alone (like the previous commit)

**Comment by nickmccurdy** [comment]
Does the current sorting algorithm still work if a table item has text that’s wrapped in multiple elements/components? If not we may want to dive recursively or sort by getting `node.textContent` for each item using refs.

**Pull request merged**